spec.html

```diff
@@ -24179,16 +24361,356 @@ <h1>Runtime Semantics: Evaluation</h1>
+      </dl>
+      <emu-alg>
+        1. Let _addInitializerClosure_ be a new Abstract Closure with parameter (_initializer_) that captures _initializers_ and performs the following steps when called:
+          1. Append _initializer_ to _initializers_.
+          1. Return *undefined*.
+        1. Let _addInitializer_ be ! CreateBuiltinFunction(_addInitializerClosure_, 1, "", &laquo; &raquo;).
+        1. Return _addInitializer_.
+      </emu-alg>
+    </emu-clause>
+
+    <emu-clause id="sec-createsetmetadatafunction" type="abstract operation">
+      <h1>
+        CreateSetMetadataFunction (
+          _kind_: ~constructor~, ~public~, or ~private~,
+          _metadataList_: a List of Metadata Records,
+          _propertyKey_: a String, Symbol, Private Name, or ~empty~,
+        ): a function object
+      </h1>
+      <dl class="header">
+        <dt>description</dt>
+        <dd>It returns a function that can be used to define metadata on the passed in _metadataList_ which is associated with the given _kind_ and, if provided, _propertyKey_.</dd>
+      </dl>
+      <emu-alg>
+        1. Let _setMetadataClosure_ be a new Abstract Closure with parameter (_metadataKey_, _value_) that captures _kind_, _metadataList_, and _propertyKey_ and performs the following steps when called:
+          1. If _metadataKey_ is not a Symbol, throw a *TypeError* exception.
+          1. Let _metadataForKey_ be the element of _metadataList_ whose [[Key]] is the same as _metadataKey_.
+          1. If _metadataForKey_ does not exist, then
+            1. Set _metadataForKey_ to a new Metadata Record { [[Key]]: _metadataKey_, [[Constructor]]: ~empty~, [[Public]]: &laquo; &raquo;, [[Private]]: &laquo; &raquo; }.
+            1. Append _metadataForKey_ to _metadataList_.
+          1. If _kind_ is ~constructor~, set _metadataForKey_.[[Constructor]] to _value_.
+          1. If _kind_ is ~public~, then
+            1. Let _publicMetadata_ be the element of _metadataForKey_.[[Public]] whose [[Key]] is the same as _propertyKey_.
+            1. If _publicMetadata_ does not exist, then
+              1. Set _publicMetadata_ to the Record { [[Key]]: _propertyKey_, [[Value]]: _value_ }.
+              1. Append _publicMetadata_ to _metadataForKey_.[[Public]].
+            1. Else, set _publicMetadata_.[[Value]] to _value_.
+          1. If _kind_ is ~private~, then
+            1. Let _privateMetadata_ be the element of _metadataForKey_.[[Private]] whose [[Key]] is the same as _propertyKey_.
+            1. If _privateMetadata_ does not exist, then
+              1. Set _privateMetadata_ to the Record { [[Key]]: _propertyKey_, [[Value]]: _value_ }.
+              1. Append _privateMetadata_ to _metadataForKey_.[[Private]].
+            1. Else, set _privateMetadata_.[[Value]] to _value_.
+          1. Return *undefined*.
+        1. Let _setMetadata_ be ! CreateBuiltinFunction(_setMetadataClosure_, 2, *""*, &laquo; &raquo;).
+        1. Return _setMetadata_.
```

Comment:
```suggestion
        1. Return CreateBuiltinFunction(_setMetadataClosure_, 2, *""*, &laquo; &raquo;).
```
(all  CreateBuiltinFunction invocations no longer return a completion record, and thus should not have a ! prefix)